### PR TITLE
Improve lyrics.modules.update_translations

### LINKED ISSUE #47: lyrics.modules.update_translations broken on new versions
Don't have time to open a PR for this now but this is broken:
- https://github.com/camptocamp/anthem/blob/master/anthem/lyrics/modules.py#L29

`update_translations` was available up to 10.0:
 - https://github.com/odoo/odoo/blob/10.0/odoo/addons/base/module/module.py#L770

`_update_translations` is available from 11.0:
 - https://github.com/odoo/odoo/blob/11.0/odoo/addons/base/module/module.py#L862
 - https://github.com/odoo/odoo/blob/12.0/odoo/addons/base/models/ir_module.py#L872
 - https://github.com/odoo/odoo/blob/master/odoo/addons/base/models/ir_module.py#L876

Maybe we should add a check to get the existing function before calling it?

### PR DESCRIPTION
**Pull request by i-vyshnevska**
remove loop in method
 bather type check

closes #47

### CODE REVIEW

**Comment by i-vyshnevska** [comment]
@simahawk review?

**Comment by simahawk** [comment]
@i-vyshnevska can you cherry pick this fix? https://github.com/camptocamp/anthem/pull/39/files#diff-a1f4dc58a2806dbab3013a09f0dac6aeR27

**Comment by i-vyshnevska** [comment]
done

File: anthem/lyrics/modules.py [hunk 1]
```diff
@@ -18,12 +18,22 @@ def uninstall(ctx, module_list):
...
                           "module's name to update the translations")
+    ir_module = ctx.env['ir.module.module']
+    if hasattr(ir_module, 'update_translations'):
+        # Odoo version <= 10.0
+        method_name = 'update_translations'
+    else:
+        # Odoo version >= 11.0
+        method_name = '_update_translations'
 
-    for module in module_list:
```
Comment (sebalix) [inline anthem/lyrics/modules.py]: Your commit removed the iteration in order to call the method on the recordset directly, and the @p-tombez one re-integrates the iteration. Is there a reason for one or the other form?

Comment (i-vyshnevska) [reply anthem/lyrics/modules.py]: I missed PR of @p-tombez in first, 
we need iteration to be able use getattr and call method in dependency of version, if method name would be same we would call it on recordset

Comment (sebalix) [reply anthem/lyrics/modules.py]: So we should be able to use `getattr` on the whole recordset, no?
```python
>>> getattr(mods, method_name)()
```

Comment (i-vyshnevska) [reply anthem/lyrics/modules.py]: yes, you are correct, updated

**Comment by i-vyshnevska** [comment]
@guewen can you check this one, seems ready for merge?

File: anthem/lyrics/modules.py [hunk 2]
```diff
@@ -18,12 +18,21 @@ def uninstall(ctx, module_list):
         raise AnthemError(u'Cannot uninstall modules. See the logs')
 
 
-def update_translations(ctx, module_list):
+def update_translations(ctx, module_list, overwrite=False):
```
Comment (guewen) [inline anthem/lyrics/modules.py]: This is a big non-backward compatible change, the `overwrite` default value was `True` before.
I think it makes more sense to have a `False` value by default so a dev has to explicitly enable `overwrite`.
But you should add a big, bold warning in the releases notes about it :)

Comment (i-vyshnevska) [reply anthem/lyrics/modules.py]: what about make it without default value, so all will be forced choose explicitly?

Comment (simahawk) [reply anthem/lyrics/modules.py]: good catch, I expected it to be False by default even before...
+1 for prominent release note + log its value w/ ctx.log_line

File: HISTORY.rst [hunk 3]
```diff
@@ -10,6 +10,8 @@ Unreleased
```
Comment (grindtildeath) [inline HISTORY.rst]: - BREAKING: Change default `overwrite` value for ``lyrics.modules.update_translations`` to False

File: anthem/lyrics/modules.py [hunk 4]
```diff
@@ -18,12 +18,23 @@ def uninstall(ctx, module_list):
...
     """ Update translations from module list"""
-    if not module_list:
+    if not isinstance(module_list, list):
         raise AnthemError(u"You have to provide a list of "
                           "module's name to update the translations")
+    if overwrite:
+        ctx.log_line(u'All previous translations will be dropped for requested addons')
```
Comment (simahawk) [inline anthem/lyrics/modules.py]:
```suggestion
        ctx.log_line(
            u'All previous translations will be dropped for requested addons'
        )
```

Comment (simahawk) [reply anthem/lyrics/modules.py]: this fixes linting failure

File: HISTORY.rst [hunk 5]
```diff
@@ -7,9 +7,11 @@ Unreleased
 ----------
 
 **Features**
-
+ - BREAKING: Change default `overwrite` value for ``lyrics.modules.update_translations`` to False
```
Comment (simahawk) [inline HISTORY.rst]:
```suggestion
 * BREAKING: Change default `overwrite` value for ``lyrics.modules.update_translations`` to False

```

File: HISTORY.rst [hunk 6]
```diff
@@ -7,9 +7,11 @@ Unreleased
...
 **Bugfixes**
 
+- Make ``lyrics.modules.update_translations`` Odoo >= 11.0 compatible
```
Comment (simahawk) [inline HISTORY.rst]:
```suggestion
* Make ``lyrics.modules.update_translations`` Odoo >= 11.0 compatible

```

**Comment by simahawk** [comment]
@i-vyshnevska please run these checks before submitting https://github.com/camptocamp/anthem/blob/master/.travis.yml#L40-L41

**Comment by i-vyshnevska** [comment]
@simahawk green, ready to merge

**Pull request merged**